RELEASE_PROCEDURE.md

```diff
@@ -1,15 +1,79 @@
-# Release Procedure
+# Release Procedures
 
-The release procedure below can be performed by a project member with
-"maintainer" or higher privileges on the GitHub repository. It assumes that you
+The release procedures below can be performed by a project member with
+"maintainer" or higher privileges on the GitHub repository. They assume that you
 will be working in an up-to-date local clone of the GitHub repository, where the
 `upstream` remote points to `github.com/sylabs/singularity`.
 
+There are two different release procedures:
+
+* A new patch release, e.g. releasing 3.11.2 when the current release is 3.11.1,
+  follows a short procedure that can be completed within a day.
+* A new major or minor version, e.g. releasing 3.12.0 when the current release
+  is 3.11.1, requires a longer procedure that takes a minimum of 2 weeks. This
+  includes a release candidate period, creation of a new release branch, new
+  documentation branches etc.
+
+## Patch Releases
+
+A patch release is made from an existing `release-XXX` branch, to provide fixes
+for the current stable major.minor version of SingularityCE.
+
+### Prior to Patch Release
+
+1. Set a target date for the patch release.
+1. Use a GitHub milestone to track issues and PRs that will form part of the
+   patch release.
+1. Cherry pick fixes from the `main` branch onto the `release-XXX` branch.
+1. Ensure that the `CHANGELOG.md` is kept up-to-date on the `release-XXX`
+   branch, with all relevant changes listed under a "Changes Since Last Release"
+   section.
+1. Monitor and merge dependabot updates, such that a release is made with as
+   up-to-date versions of dependencies as possible. This lessens the burden in
+   addressing patch release fixes that require dependency updates, as we use
+   several dependencies that move quickly.
+1. Identify and manually test any changes that are not exercised by the CI
+   pipeline. E.g. cgroups v1 code.
+
+### Creating the Patch Release
+
+1. Check the version of Go installed in the CI environment from
+   `.circleci/config.yml`. Ensure that it is the latest release, as it will be
+   used to build our release packages. Update if necessary via a PR against the
+   release branch.
+1. Modify the `README.md`, `INSTALL.md`, `CHANGELOG.md` via PR against the
+   release branch, so that they reflect the version to be released.
+1. After all PRs are merged, apply an annotated tag to the HEAD of the release
+   branch via `git tag -a -m "SingularityCE v3.11.2" v3.11.2`.
```

Comment:
Might be worth signing the release tags?
```suggestion
   branch via `git tag -a -s -m "SingularityCE v3.11.2" v3.11.2`.
```

Reply:
Holding on this pending possible strategy for cross-siging / publishing keys.